### PR TITLE
[Docs] Add: register offline data as a feature set; feature store spark using Redis; SQL source, target and driver

### PR DESCRIPTION
**Pull request by jillnogold**
ML-2592, ML-3008, ML-2610

### CODE REVIEW

File: docs/feature-store/feature-sets.md [hunk 1]
```diff
@@ -43,13 +43,33 @@ Example:
 stocks_set = FeatureSet("stocks", entities=[Entity("ticker")])
 ```
 
-To learn more about Feature Sets go to {py:class}`~mlrun.feature_store.FeatureSet`.
 
-```{admonition} Note 
-Feature sets can also be created in the UI. To create a feature set:
+### Create a feature set without ingesting its data
+
+You can define and register a feature set, and use it in a feature vector, without ingesting its data. This method is
```
Comment (assaf758) [inline docs/feature-store/feature-sets.md:49]: You can define and register a feature set (and use it in a feature vector) without ingesting its data into mlrun offline targets.
The use-case for this is when you have a large amount of data at some remote storage that is ready to be consumed by a model-training pipeline.
When this feature is enabled on a feature-set, data is not saved to offline target during ingestion. Instead, when a get_offline_features is called on a vector containing that feature-set, that data will be read directly from the source.
Note that online targets are still ingested, and their value represents a time-slice of the offline source.
Note also that transformations are not allowed when this feature is enabled: No computation graph, no aggregations etc.
Enable this feature by including `passthrough=True` in the feature set definition. All three ingestion engines (Storey, Spark, Pandas) are supported, as well as retrieval engines "local" and "spark".

File: docs/data-prep/ingest-data-fs.md [hunk 2]
```diff
@@ -187,3 +213,24 @@ For example: `rediss://:abcde@localhost:6379` creates a redis target, where:
...
 explicitly each time with the path parameter, for example:</br>
 `RedisNoSqlTarget(path ="redis://1.2.3.4:6379")`
+
+### SQL target store
+
+The `SQLTarget` online target supports storey but does not support spark.<br>
+To configure, pass the `db_uri` or overwrite the `MLRUN_SQL__URL` env var, in this format:<br>
+`mysql+pymysql://<username>:<password>@<host>:<port>/<db_name>`
+
+You can pass the schema of the table you want to create or the name of an existing table, for example:
```
Comment (davesh0812) [inline docs/data-prep/ingest-data-fs.md:223]:
```suggestion
You can pass the schema and the name of the table you want to create or the name of an existing table, for example:
```

File: docs/data-prep/ingest-data-fs.md [hunk 3]
```diff
@@ -187,3 +213,24 @@ For example: `rediss://:abcde@localhost:6379` creates a redis target, where:
...
+ target = SQLTarget(
+            table_name=target_name,
+            create_table=False,
+            primary_key_column=key,
+            time_fields=["time"]
+        )
+        feature_set = fs.FeatureSet(
+            "my_fs", entities=[fs.Entity('key')],
+        )
+        fs.ingest(feature_set, source=origin_df, targets=[target])
```
Comment (davesh0812) [inline docs/data-prep/ingest-data-fs.md:235]:
```suggestion
 target = SQLTarget(
            table_name='my_table',
            schema= {'id': string, 'age': int, 'time': pd.Timestamp, ...}
            create_table=True,
            primary_key_column='id',
            time_fields=["time"]
        )
feature_set = fs.FeatureSet("my_fs", entities=[fs.Entity('id')],)
fs.ingest(feature_set, source=df, targets=[target])
```

File: docs/data-prep/ingest-data-fs.md [hunk 4]
```diff
@@ -146,16 +146,42 @@ For Spark engine, other sources are also supported.
...
+       source = SqlDBSource(
+           table_name='source_name', db_path=self.db, key_field='key', time_fields=['timestamp'],
+       )
+
+
+        feature_set = fs.FeatureSet(
+            "my_fs", entities=[fs.Entity('key')],
+        )
+        feature_set.set_targets([])
+        df = fs.ingest(feature_set, source=source)
```
Comment (davesh0812) [inline docs/data-prep/ingest-data-fs.md:180]:
```suggestion
source = SqlDBSource(table_name='my_table', 
                     db_path="mysql+pymysql://abc:abc@localhost:3306/my_db", 
                     key_field='key',
                     time_fields=['timestamp'], )
 
 feature_set = fs.FeatureSet("my_fs", entities=[fs.Entity('key')],)
 feature_set.set_targets([])
 df = fs.ingest(feature_set, source=source)
```

**Pull request merged**